src/main/scala/centaur/test/workflow/WorkflowData.scala

```diff
@@ -3,13 +3,38 @@ package centaur.test.workflow
 import java.nio.file.Path
 
 import cats.data.Validated._
+import centaur.CentaurConfig
 import centaur.test._
 import com.typesafe.config.Config
 import configs.Result
 import configs.syntax._
+import spray.json._
+
 
 case class WorkflowData(wdl: String, inputs: Option[String], options: Option[String])
 
+case class WorkflowOptions(options: Option[String]) {
+  import DefaultJsonProtocol._
+
+  def insertSecrets(): Option[String] = {
+    val tokenKey = "refresh_token"
+
+    def addToken(optionsMap: Map[String, JsValue]): Map[String, JsValue] = {
+      CentaurConfig.optionalToken match {
+        case Some(token) if optionsMap.get(tokenKey).isDefined => optionsMap - tokenKey + (tokenKey -> JsString(token))
+        case None => optionsMap
```

Comment:
Be careful here, if you have Some(token) but optionsMap.get(tokenKey) is *not* defined you'll get a match error. You probably want `case _ => optionsMap` if I understand what you're doing correctly